src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,109 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.TodoCommand;
+import duke.command.UnmarkCommand;
+
+/**
+ * Class that parses user input and returns the corresponding command.
+ */
+public class Parser {
+
+    /**
+     * Main method that takes in raw user input and returns the corresponding command.
+     *
+     * @param fullCommand The raw string input by the user.
+     * @return The command corresponding to the user's input.
+     * @throws DukeException If user's input is invalid (does not correspond to any command or invalid parameters).
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] inputs = fullCommand.split(" ", 2); //return [commandWord, restOfCommand]
+        String commandWord = inputs[0];
+
+        switch(commandWord) {
+        case ListCommand.COMMAND_WORD:
+            return new ListCommand();
+        case MarkCommand.COMMAND_WORD: {
+            int index = -1;
+            try {
+                index = Integer.valueOf(inputs[1]) - 1;
+                return new MarkCommand(index);
+            } catch (NumberFormatException e) {
+                throw new DukeException("Invalid parameter(s). Usage: mark [taskNumber]");
+            } catch (IndexOutOfBoundsException e) {
+                throw new DukeException("Invalid parameter(s). duke.task.Task " + (index + 1) + " does not exist");
```

Comment:
I think `IndexOutOfBoundsException` will never be caught here? The error will be thrown in the `execute` function and the same error was handled there as well.